src/core/assertions.ts

```diff
@@ -105,149 +88,146 @@ exports.countNestedProperties = (object) => {
  *     ]
  * )
  * // Will throw because last_name does not equal 'Dupond'.
- *
- * @see ObjectFieldSpec
- *
- * @param {object}            object        - object to test
- * @param {ObjectFieldSpec[]} spec          - specification
- * @param {boolean}           [exact=false] - if `true`, specification must match all object's properties
+ * [exact=false] - if `true`, specification must match all object's properties
  */
-exports.assertObjectMatchSpec = (object, spec, exact = false) => {
-    spec.forEach(({ field, matcher, value }) => {
-        const currentValue = _.get(object, field)
-        const expectedValue = Cast.value(value)
 
-        const rule = exports.getMatchingRule(matcher)
+export const assertObjectMatchSpec = (
+    object: Record<string, unknown>,
+    spec: ObjectFieldSpec[],
+    exact = false
+): void => {
+    for (const { field, matcher, value } of spec) {
+        const currentValue = getValue(object, field) as string
+        const expectedValue = Cast.getCastedValue(value) as string
+
+        const rule = getMatchingRule(matcher)
+        if (!rule) return
 
+        let message: string
```

Comment:
```suggestion
        const message = (msg: string) => `Property '${field}' (${currentValue}) ${msg} '${expectedValue}'`
```

Reply:
+ example in the next comment (to reproduce on each RuleName)

Reply:
Yes, thank you for suggesting this change.